api.lua

```diff
@@ -63,9 +63,16 @@ function mail.send(m)
 		m.from, m.to, extra_log, m.subject, m.body
 	))
 
+
+	local id = mail.new_uuid()
+	if m.id then
+		mail.delete_mail(m.from, m.id)
+		id = m.id
+	end
+
 	-- form the actual mail
 	local msg = {
-		id = mail.new_uuid(),
+		id = id,
```

Comment:
Minor thing but here you could easily save call to `mail.new_uuid()` when it is not necessary / result not used:
```lua
	local id
	if m.id then
		mail.delete_mail(m.from, m.id)
		id = m.id
	end

	-- form the actual mail
	local msg = {
		id = id or mail.new_uuid(),
```